src/kinesis.rs

```diff
@@ -92,18 +96,31 @@ where
                 }
                 None => {
                     self.get_config()
-                        .tx_records
-                        .send(Err(ProcessError::PanicError(
-                            "ShardIterator is None".to_string(),
-                        )))
+                        .tx_ticker_updates
+                        .send(TickerMessage::RemoveShard(res.shard_id.clone()))
                         .await
-                        .expect("");
+                        .expect("Could not send RemoveShard to tx_ticker_updates");
+                    rx_shard_iterator_progress.close();
```

Comment:
Also handling shards that have "terminated", usually because of a scaling operation.